### ISSUE TITLE
IPython walk-through?

### ISSUE THREAD

**Issue by klonuo** [pgbovine/OnlinePythonTutor#146]
Hi,

although this seem like a great project, at first to me the code base looks like a mess - different versions in one package with a sparse documentation unrelated to each other and a mix of different files and folders without clear purpose (to me). 

I first saw it on a short YouTube video in action, without any pointers how to get there. IPython doesn't list it as extension and even Google could not find me any walk-through, while ironically this package should target noobs.

What I did and hope that there must be a better way, is that I just copied these files:

```
v3/callback_module.py
v3/html_module.py
v3/htmlexample_module.py
v3/htmlFrame.py
v3/matrix.py
v3/opt-ipy.py
v3/pg_encoder.py
v3/pg_logger.py
v3/ttt_module.py
```

to my IPython extensions.

Assuming I'm addressing right place, could someone please write a walk-through and link it on a front page?

**Comment by techtonik** [comment]
I guess you're the best person to write this walk-through, because, frankly, I don't understand what do you want to accomplish.

**Comment by klonuo** [comment]
Hm... I just got introduced to this package today. 

What I wanted to accomplish is of course to make this package work for me as shown in that short YouTube video linked in `opt-ipy.py` comment. 

I was surprised that there is no OPT extension listed in [IPython extensions](https://github.com/ipython/ipython/wiki/Extensions-Index) and I copied the files based on import dependencies, which I don't consider trivial, and wasn't sure if that's recommended way to use it.

**Comment by techtonik** [comment]
Yes, OPT codebase is somewhat of a flux, because main module is the site, not addon for IPython. So if you've got the walk-through, it is worth to publish it and think about improving the process later. IPython example is very good.

**Comment by klonuo** [comment]
Ok, on it

**Comment by klonuo** [comment]
Initial commit: https://github.com/klonuo/opt-ipy

I'll clean the code next, as for example `pg_logger.py` contains mixed indentation, and perhaps update IPython extension to target IPython 2 or later.

Any suggestions, as where to next, are welcome

**Comment by techtonik** [comment]
Try to make it a clone of original repo, so that it could be merged back here.

**Comment by klonuo** [comment]
That was the idea, but I don't think it is possible. I tried adding OPT as git submodule to my repo, then symlink selected files from this submodule and updating `.gitmodules`, but it didn't work for me as expected.

Do you have other idea?

**Comment by lrq3000** [comment]
This should be made part of the standard Jupyter package. This has a BIG potential for step-by-step visual debugging, with a little more work breakpoints could be set and live variables modification could be added. But as it is, it would be tremendously helpful for new Python programmers, this is a killer feature.

**Comment by lrq3000** [comment]
Addendum: partcularly if it could be implemented in an IPython Notebook, the visual approach would perfectly fit! Like eg the possibility to run a cell in the visualizer!
